lnst/Tests/LinuxPerf.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional
+import subprocess
+import logging
+import re
+import os
+
+from lnst.Tests.BaseTestModule import BaseTestModule
+from lnst.Common.Parameters import StrParam, IntParam, ListParam
+from lnst.Common.Utils import is_installed
+from lnst.Common.ExecCmd import log_output
+
+class LinuxPerf(BaseTestModule):
+    output_file = StrParam(mandatory=True)
+    cpus = ListParam(type=IntParam())
+    events = ListParam(type=StrParam())
+
+    def run(self) -> bool:
+        self._res_data = {}
+        if not is_installed("perf"):
+            self._res_data["msg"] = "perf is not installed on this machine!"
+            logging.error(self._res_data["msg"])
+            return False
+
+        # can't use lnst.Common.ExecCmd.exec_cmd directly, because expected returncode is not zero
+        cmd: str = self._compose_cmd()
+        logging.debug(f"Executing: \"{cmd}\"")
+        process = subprocess.Popen(
+            cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True
+        )
+
+        self.wait_for_interrupt()
+
+        stdout, stderr = process.communicate()
+
+        if stdout:
+            log_output(logging.debug, "Stdout", stdout.decode())
+        if stderr:
+            log_output(logging.debug, "Stderr", stderr.decode())
+
+        self._res_data["filename"] = os.path.abspath(self._parse_filename_from_output(stderr.decode()))
```

Comment:
Could you please explain why we need this when the `output_file` is a mandatory parameter? Can't we use it instead?

Reply:
I just now realized, that I'll still need to pass the filename through `_res_data["fileame"]`, because when copying the result file from agent to controller, we'd rather have an absolute path, just to be sure. Don't know if agent executes rpc calls from the same directory.

Reply:
I removed `--timestamp-filename` and use `self.parsed.output_file` directly.